services/bots.py

```diff
@@ -77,21 +97,23 @@ def _get_apps_from_weighted_selection(
     # Obviously apps with a higher weight have a higher change of being selected as the main app (1st selection)
     # But it's important that others are also selected so we can use them as fallbacks
     apps_to_consider = []
-    if keys:
-        while len(keys) > 1:
-            selected_app_id = random.choices(keys, weights, k=1)[0]
-            apps_to_consider.append(ghapp_installations_filter[selected_app_id])
-            # random.choices chooses with replacement
-            # which we are trying to avoid here. So we remove the key selected and its weight from the population.
-            key_idx = keys.index(selected_app_id)
-            keys.pop(key_idx)
-            weights.pop(key_idx)
-        # The app remaining is the last choice
-        apps_to_consider.append(ghapp_installations_filter[keys[0]])
+    apps_to_select = len(keys)
+    selections = 0
+    while selections < apps_to_select:
+        selected_app_id = random.choices(keys, weights, k=1)[0]
+        apps_to_consider.append(ghapp_installations_filter[selected_app_id])
+        # random.choices chooses with replacement
+        # which we are trying to avoid here. So we remove the key selected and its weight from the population.
+        key_idx = keys.index(selected_app_id)
+        keys.pop(key_idx)
+        weights.pop(key_idx)
+        selections += 1
     if installation_name != GITHUB_APP_INSTALLATION_DEFAULT_NAME:
         # Add the default app as the last fallback if the owner is using a different app for the task
         default_apps = filter(
-            lambda obj: obj.name == GITHUB_APP_INSTALLATION_DEFAULT_NAME,
+            lambda obj: _can_use_this_app(
```

Comment:
Just double checking that this is OK. `_can_use_this_app` has extra logic that's not in the original functionality of this code, but those extra checks are required to be able to use the app anyways(?)

Reply:
I actually noticed when refactoring that the repo verification was missing from this filter. I think it is necessary to have that too, because if the repo is not covered we can't use the app.

Of course, considering the default installation, we expect all repos (that Codecov has access to) to be covered, but it's better for the logic to be consistent.

Reply:
About the `is_configured` check is a similar story. We generaly expect the default app to be configured (otherwise it wouldn't have that name actually), but if it's not configured we shouldn't use it.